src/Logging/Logging.Abstractions/src/EventId.cs

```diff
@@ -50,6 +50,27 @@ public EventId(int id, string name = null)
             Name = name;
         }
 
+        /// <summary>
+        /// Initializes an instance of the <see cref="EventId"/> struct.
+        /// The numeric identifier is derived from the given name.
+        /// </summary>
+        /// <param name="name">The name of this event.</param>
+        public EventId(string name)
+        {
+            unchecked
+            {
+                var hash = 3074457345618258791ul;
+                for (var index = 0; index < name.Length; ++index)
+                {
+                    hash += name[index];
+                    hash *= 3074457345618258799ul;
+                }
+
+                Id = 100000 + (int)(hash % 900000);
```

Comment:
I don't like crossing our fingers and hoping Id's don't collide. If developers are willing to accept the risk of collisions, we might as well use 0 as the Id, so that developers are forced to deal with the colliding Id's early.

Reply:
Although that's true, a lot of teams have devs who are happy to provide reasonable event names (especially if the string eventName isn't optional) but will balk at the the suggestion they do manual book-keeping needed to assign a unique ids to every event name in a category. Providing a reasonably unique EventId.Id for free is a way to get most of the benefits of the Id field that doesn't burden the dev teams.

I had the same concern about collisions at first also, but ended up thinking about it like this... Just like you implied even if the EventId.Id has a 100% collision rate the EventId.Name property are still different and message types can still be distinguished. Sinks like Serilog will already be storing EventId.Id and EventId.Name. It would probably also be a good idea for console and debug logger to display EventId.Name when provided instead of (or in addition to) the EventId.Id.

To quantify the odds of collision just did some simulations. Random two-part event names made from a 7700-word dictionary. Using a 9-digit Id, and assuming 10 event names each class category names, got 6 collisions over 100 million class category names.

https://gist.github.com/lodejard/725e849fa9ff57b01a10e07996d8dc16